src/cryptsetup/cryptsetup-generator.c

```diff
@@ -262,6 +262,7 @@ static int create_disk(
                 "RemainAfterExit=yes\n"
                 "TimeoutSec=0\n" /* the binary handles timeouts anyway */
                 "KeyringMode=shared\n" /* make sure we can share cached keys among instances */
+                "OOMScoreAdjust=1000\n" /* unlocking can allocate a lot of memory if Argon2 is used */
```

Comment:
hmm, why so agressive? I think there are plenty of usecases where other stuff should be even more likely to be called by the OOM killer. I think 500 should suffice too. It makes it very likely these services are killed on OOM pressure, but leaves room for other stuff.

Reply:
Well, if the old key derivation function is used then OOM is very unlikely to happen in early boot. And if Argon2 is used then it is almost certain that systemd-cryptsetup is the offender.

But sure, we don't have to be so aggressive.

Reply:
we never know what might run in parallel here... after all these things are just relative. and I think we should change the defaults here, as your patch does, but still leave room for downstreams to order their own stuff against it.

Reply:
No problem, we can always adjust later. Btw, I've just force-pushed new version (dropping needs-rework label).